python/tvm/driver/tvmc/autotuner.py

```diff
@@ -228,24 +240,137 @@ def drive_tune(args):
+        args.trials,
+        args.target_host,
+        args.tuner,
+        args.min_repeat_ms,
+        args.early_stopping,
+        args.desired_layout,
+        args.timeout,
+        args.number,
+        args.repeat,
+        args.parallel,
+        hardware_params,
+        args.include_simple_tasks,
+        args.log_estimated_latency,
+    )
+
+
+def tune_model(
+    tvmc_model: TVMCModel,
+    target: str,
+    tuning_records: Optional[str] = None,
+    enable_autoscheduler: bool = False,
+    rpc_key: Optional[str] = None,
+    rpc_tracker: Optional[str] = None,
+    trials: Optional[int] = None,
+    target_host: str = "llvm",
+    tuner: str = "xgb",
+    min_repeat_ms: Optional[int] = None,
+    early_stopping: Optional[int] = None,
+    desired_layout: Optional[str] = None,
+    timeout: int = 10,
+    number: int = 10,
+    repeat: int = 1,
+    parallel: int = 4,
+    hardware_params: Optional[HardwareParams] = None,
+    include_simple_tasks: bool = False,
+    log_estimated_latency: bool = False,
+):
+    """Use tuning to automatically optimize the functions in a model.
+
+    Parameters
+    ----------
+    tvmc_model : TVMCModel
+        The model to be optimized.
+    target : str
+        Compilation target as plain string, inline JSON or path to a JSON file.
+    tuning_records: str, optional
+        The path to a file that tuning results will be saved to. If not specified,
+        a temporary file will be used.
+    enable_autoscheduler : bool, optional
+        When true, use autoscheduling rather than autotvm. This should produce
+        faster kernels for compatible model-target pairs.
+    rpc_key : str, optional
+        The RPC tracker key of the target device. Required when rpc_tracker is provided.
+    rpc_tracker : str, optional
+        The hostname and port (optional, defaults to 9090) of the RPC tracker,
+        e.g. 192.168.0.100:9999.
+    trials : int, optional
+        The number of schedules to try out. For autotvm, each task will have this many
+        options explored. For autoscheduling, the total number of schedules checked in
+        the entire model will be this many.
+    target_host : str, optional
+        The host compilation target, defaults to 'llvm'.
+    tuner : str, optional
+        The type of tuner to use when tuning with autotvm. Can be one of
+        "ga", "gridsearch", "random", "xgb", "xgb_knob", and "xgb-rank".
+    min_repeat_ms : int, optional
+        Minimum time to run each trial. Defaults to 0 on x86 and 1000 on other targets.
+    early_stopping : int, optional
+        When specified, stop tuning after this number of trials if results aren't improving.
+    desired_layout : str, optional
+        Can be one of "NCHW" or "NHWC". When specified, the graph will be converted to this layout.
+    timeout : int, optional,
+        If a kernel trial lasts longer than this duration in seconds, it will be
+        considered a failure.
+    number : int, optional
```

Comment:
I agree this might be useful. I would like to hold off on it as it would go well with a PR aimed at simplifying TVMC arguments through the use of more structs and JSON.

Reply:
I guess to play devil's advocate: this PR is introducing a top-level API. why is that out of scope now? I understand that it might be hard to do all of that plus e.g. yaml considerations, but organization seems like a fundamental part of this PR.

Reply:
The main reason I'd like to separate into another PR is that using more structs / JSON would be very beneficial to command line TVMC as well. I don't want this PR to be disruptive to the command line interface since that is used in real applications already. Anything that significantly effects that flow should be a smaller PR imo.

Reply:
I don't think you have to modify the command-line interface to handle JSON in order to organize params at this level. I think "number" and "trials" are extremely confusing as kwargs to the same API.

Reply:
I agree with you and we will definitely do this but dont want to add it to this PR, it's already too big.